### ISSUE TITLE
`excluded` seems to be ignore in 0.65.0

### ISSUE THREAD

**Issue by michaeldclifford** [realm/SwiftLint#6795]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

In **0.64.1**, `swiftlint` passes.
In **0.65.0**, `swiftlint` reports thousands of violations, all in folders that have been filed under `excluded` in the `swiftlint.yml` file

```bash
$ swiftlint lint
```

### Environment

* SwiftLint version: working in **0.64.1**, not working in **0.65.0**
* Xcode version: 26.6 (17F113)
* Installation method used: CocoaPods
* Configuration file:

```yml
excluded:
  - DerivedData
  - Pods
```

**Comment by SimplyDanny** [comment]
Can you provide a simple setup that shows the difference?

A concrete example would be very helpful, as all tests we have in place passed. Perhaps there are symbolic links or special paths involved in your setup.

**Comment by pganaus** [comment]
I can confirm that in some of my targets `excluded` breaks starting with version `0.64.0`. So far I couldn't figure out what the difference between the targets is.

My folder structure looks like:

- Workspace.xcworkspace
- .swiftlint.yml
- Tools
  - SwiftLint
    - swiftlint
- ProjectA
  - ProjectA.xcodeproj
  - Common
  - ToBeExcluded
  - TargetX
  - TargetY
  - TargetZ
- ProjectB
- ...

`swiftlint` is called as a script phase in the targets build:
```
if [[ "$CONFIGURATION" = "Debug" || "$CONFIGURATION" = "Debug Test" ]]; then
  if [[ -e $PROJECT_DIR/../Tools/SwiftLint/swiftlint ]]; then
    $PROJECT_DIR/../Tools/SwiftLint/swiftlint lint --config $PROJECT_DIR/../.swiftlint.yml --use-script-input-file-lists
  else
    echo "warning: SwiftLint not available!"
  fi
fi
```

The `excluded` in the `swiftlint.yml` looks like this:
```
excluded: # Paths to exclude from linting
  - ProjectA/ToBeExcluded
```

Environment

    SwiftLint version: working in 0.63.3, not working starting with 0.64.0
    Xcode version: 26.3 (17C529)
    Installation method used: Using the portable swiftlint

**Comment by oscamv-verkada** [comment]
Maybe related: If I recreate the baseline for my project it'll change all the file paths into full paths instead of from the root directory.

**Comment by tomotakatakahashi** [comment]
Hi. I am experiencing an issue similar to the one reported in  [pganaus's comment](https://github.com/realm/SwiftLint/issues/6795#issuecomment-4862623072) (and likely different from [michaeldclifford's original post](https://github.com/realm/SwiftLint/issues/6795#issue-4768619129)).

## Summary

* Running `swiftlint` directly from the terminal succeeds. (v0.65.0 from Homebrew)
* Running `pre-commit run -a` fails with v0.64.0 and v0.65.0, but works fine with v0.63.3.
  * More specifically, `792032480e5558a0cd490dbd25d37ced0eeda804` fails and `792032480e5558a0cd490dbd25d37ced0eeda804~` passes.

```
$ pre-commit run -a
SwiftLint................................................................Failed
- hook id: swiftlint
- exit code: 2

/${this_repo}/MyPackage/Package.swift:1:5: error: Prefixed Top-Level Constant Violation: Top-level constants should be prefixed by `k` (prefixed_toplevel_constant)
```

## Reproduction Setup

`.pre-commit-config.yaml`
```yaml
repos:
  - repo: https://github.com/realm/SwiftLint
    rev: 0.65.0
    hooks:
      - id: swiftlint
```

`.swiftlint.yml`

```yaml
strict: true

excluded:
  - MyPackage/Package.swift

opt_in_rules:
  - prefixed_toplevel_constant
```

`NoProblem.swift`

```swift
// Just to avoid "No lintable files found" errors.
func incl(_ value: Int) -> Int { value + 1 }
```

`MyPackage/Package.swift`

```swift
let package = "Package(...) actually."
```

---
Thanks for maintaining this project 🙂

**Comment by alexanderwe** [comment]
I am seeing a similar issue starting with `0.64.0`. When using `--use-script-input-file-lists` in a Xcode build script the paths seem to be not compared against the `excluded` file paths causing errors to be reported for files that should be excluded/ignored. 
Not sure which commit causes the regression. I am not sure though if the previous behaviour was correct and if in general the `excluded` rule should not trigger when using `--use-script-input-file-lists`.

**Comment by lilidotshi** [comment]
Would like to bump on this because we are also running into this issue. The work around we've found is to specify --force-exclude
